src/cli.rs

```diff
@@ -141,6 +147,272 @@ pub(crate) fn parse_startup_args() -> Result<LdkUserInfo, ()> {
 	})
 }
 
+// Default datadir relative to home directory
+#[cfg(target_os = "windows")]
+const DEFAULT_BITCOIN_DATADIR: &str = "AppData/Bitcoin";
+#[cfg(target_os = "linux")]
+const DEFAULT_BITCOIN_DATADIR: &str = ".bitcoin";
+#[cfg(target_os = "macos")]
+const DEFAULT_BITCOIN_DATADIR: &str = "Library/Application Support/Bitcoin";
+
+// Environment variable/.env keys
+const BITCOIND_RPCUSER_KEY: &str = "RPCUSER";
+const BITCOIND_RPCPASSWORD_KEY: &str = "RPCPASSWORD";
+
+fn print_rpc_auth_help() {
+	println!("To provide the bitcoind RPC username and password, you can either:
+1. Provide the username and password as the first argument to this program in the format:
+   <bitcoind-rpc-username>:<bitcoind-rpc-password>@<bitcoind-rpc-host>:<bitcoind-rpc-port>
+2. Provide <bitcoind-rpc-username>:<bitcoind-rpc-password> in a .cookie file in the default bitcoind data directory: `$HOME/{data_dir}`
+3. Set the {rpc_user_key} and {rpc_pass_key} environment variables
+4. Provide {rpc_user_key} and {rpc_pass_key} fields in a .env file in the current directory",
+	data_dir = DEFAULT_BITCOIN_DATADIR, rpc_user_key = BITCOIND_RPCUSER_KEY, rpc_pass_key = BITCOIND_RPCPASSWORD_KEY);
+}
```

Comment:
Would have liked to have dynamically found the default directory here:
```rust
fn print_rpc_auth_help() {
	let home_path = env::home_dir();
	let home_path = match home_path {
		Some(ref p) => p.to_str().unwrap_or("$HOME"),
		None => "$HOME"
	};
	let data_dir = home_path.to_string() + "/" + DEFAULT_BITCOIN_DATADIR;
	println!("To provide the bitcoind RPC username and password, you can either:
1. Provide the username and password as the first argument to this program in the format:
   <bitcoind-rpc-username>:<bitcoind-rpc-password>@<bitcoind-rpc-host>:<bitcoind-rpc-port>
2. Provide <bitcoind-rpc-username>:<bitcoind-rpc-password> in a .cookie file in the default bitcoind data directory: `{data_dir}`
3. Set the {rpc_user_key} and {rpc_pass_key} environment variables
4. Provide {rpc_user_key} and {rpc_pass_key} fields in a .env file in the current directory",
	data_dir = data_dir, rpc_user_key = BITCOIND_RPCUSER_KEY, rpc_pass_key = BITCOIND_RPCPASSWORD_KEY);
}
```
But for some reason it would not build in Rust 1.48.0. I tried removing/modifying the added code to see what was the problem but no matter what I did it would fail to build. And it wouldn't give me any compiler messages, just 
```
error: could not compile `ldk-sample`

Caused by:
  process didn't exit successfully: <huge command similar to what failed in [the CI on my previous commit](https://github.com/lightningdevkit/ldk-sample/actions/runs/3747465547/jobs/6367622902)>
```
Figured it was a very small thing so just left it out for now.

Reply:
Figured it out! It was not a problem with the code, just ran `cargo clean` and building worked.